core/server/services/route/settings.js

```diff
@@ -0,0 +1,74 @@
+'use-scrict';
+
+const yaml = require('js-yaml'),
+    fs = require('fs-extra'),
+    path = require('path'),
+    debug = require('ghost-ignition').debug('settings'),
+    common = require('../../lib/common'),
+    config = require('../../config'),
+    // @TODO: move to settings service
+    routesFile = path.join(config.getContentPath('settings'), 'routes.yaml'),
+    defaultRoutesFile = path.join(config.get('paths').defaultRoutes, 'default-routes.yaml');
+
+let loadSettings;
+
+// We should ALWAYS have a `routes.yaml` file available in content/settings.
+// This can be a custom file, or our default. If the file is missing, we
+// copy the default routes file back.
+function ensureRoutesFile() {
+    return fs.readFile(routesFile, 'utf8')
+        .then((file) => {
+            debug('Found routes file in settings folder');
+            return Promise.resolve(file);
+        }).catch({code: 'ENOENT'}, () => {
+            // File doesn't exist, copy it from our defaults
+            return fs.copy(defaultRoutesFile, routesFile).then(() => {
+                return fs.readFile(routesFile, 'utf8');
+            }).then((file) => {
+                debug('Default routes file copied');
+                return Promise.resolve(file);
+            }).catch((err) => {
+                // the default routes file doesn't exist 😱, or we can't access
+                // the content/settings folder. Now we have the salad
+                // TODO: what error do we want to return here
+                return Promise.reject(new common.errors.GhostError({
+                    message: 'Error trying to copy the default routes file.',
```

Comment:
The only other errors I can actually think of are 
- error with usage of a storage adapter
- missing `content/settings` directory (user deleted it, or it's not there yet)
- no permission to access the `content/settings` directory (CLI case usually)

Not sure, if you can think of any more cases...